.github/workflows/pants.yaml

```diff
@@ -27,13 +27,37 @@ jobs:
     steps:
     - uses: actions/checkout@v2
     - uses: actions/cache@v2
-      id: cache
+      id: cache_pants_setup
       with:
         path: |
           ~/.cache/pants/setup
-          ~/.cache/pants/lmdb_store
+        key: pants-setup-${{ runner.os }}-${{ hashFiles('pants.toml') }}
```

Comment:
FWIW here's what we use for finer-grained caching using just the Pants version:

```
      - name: Get Pants version
        id: pants_version
        run: |
          PANTS_VERSION=$(cat pants.toml | python3 -c 'import sys,re;m=re.search("pants_version\s*=\s*\"(\S+)\"", sys.stdin.read());print(m.group(1) if m else sys.exit(1))')
          echo "::set-output name=pants_version::$PANTS_VERSION"
      - name: Cache Pants runner
        uses: actions/cache@v3
        with:
          path: ~/.cache/pants/setup
          key: pants-setup-${{ steps.pants_version.outputs.pants_version }}
```

Reply:
We could _very easily_ provide a little GitHub Action we host just for caching the Pants setup?

Reply:
Sure, but, you know, not for this change...

Reply:
I think I've found a good compromise for this. That full incantation above is complicated and bogs the docs down, but for the purpose of invalidation there is no need to be that precise. We just need to grep out the entire `pants_version = ` line. Invalidating if anyone changes whitespace in that line is NBD.

Reply:
Yeah, let's open an issue so we can track publishing our own GHA for this.

I want our GitHub users to be just as successful (when possible) as our other CI users 🙂